src/cephadm/cephadm

```diff
@@ -2292,10 +2292,34 @@ def command_version():
 @infer_image
 def command_pull():
     # type: () -> int
-    logger.info('Pulling latest %s...' % args.image)
-    call_throws([container_path, 'pull', args.image])
+
+    _pull_image(args.image)
     return command_inspect_image()
 
+
+def _pull_image(image) -> None:
+    logger.info('Pulling container image %s...' % image)
+
+    ignorelist = [
+        "error creating read-write layer with ID",
+        "net/http: TLS handshake timeout",
+    ]
+
+    cmd = [container_path, 'pull', image]
+
+    retry = args.retry
+
+    for count in range(retry):
+        out, err, ret = call(cmd)
+        if not ret:
+            return
+
+        if not any(pattern in err for pattern in ignorelist):
```

Comment:
I'm thinking the "not" in this line should be dropped, otherwise it will raise an exception when the image pull operation succeeds.

Reply:
oh, never mind

Reply:
right. if the pull succeeds, we're escaping in `if not ret`